pool.go

```diff
@@ -138,7 +153,7 @@ func (p *Pool) Get(ctx context.Context) (*ClientConn, error) {
 	case wrapper = <-clients:
 		// All good
 	case <-ctx.Done():
-		return nil, ErrTimeout
+		return nil, ErrTimeout // it would better returns ctx.Err()
```

Comment:
I agree, but that'd also be a breaking change for clients using this constant in their error checks, so I'm keen on keeping this constant for now

Reply:
agree with you, that's the reason I didn't change it, may be fixed next API breaking change.